src/gameboard.js

```diff
@@ -0,0 +1,177 @@
+const Ship = require('./ship');
+
+function Gameboard(player = null) {
+
+  return {
+    ships: [],
+    moves: [],
+    letterToNum: {
+      'A': 0,
+      'B': 1,
+      'C': 2,
+      'D': 3,
+      'E': 4,
+      'F': 5,
+      'G': 6,
+      'H': 7,
+      'I': 8,
+      'J': 9
+    },
+    createShip: function(n) {
+      const newShip = Ship(n);
+      return newShip;
+    },
+
+    board: (function() {
+      const board = [];
+      let board_row = [];
+      let row = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J'];
+      let col = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'];
+      for (let row_element of row) {
+        for (let col_element of col) {
+          board_row.push(row_element + col_element);
+          if (board_row.length === 10) {
+            board.push(board_row);
+            board_row = [];
+          }
+        }
+      }
+      return board;
+
+    })(),
+
+    placeShip: function(ship, x, y) {
+
+      const row_x = this.letterToNum[x];
+
+      let max_length = y + ship.length;
+      if (max_length <= 10) {
+        for (let i = y; i < max_length; i++) {
+          this.board[row_x][i - 1] += " ";
+          this.board[row_x][i - 1] += `ship_${this.ships.length}`;
+
+          if (this.board[row_x - 1]) {
+            if (!this.board[row_x - 1][i - 1].match(/blank/g)) {
+              this.board[row_x - 1][i - 1] += "-blank";
+            }
+
+          }
+
```

Comment:
this your technic of implementing function bodies inside the return statement can get confusing lol. It works well, i know.
Try and organise them this manner shown in the[ ODIN PROJECT](https://www.theodinproject.com/courses/javascript/lessons/factory-functions-and-the-module-pattern)

It's easier to read, and more "industry-standard" 😊😊

(good thing they already work, you just have to organise)